File: multiStructure/frameworks/csharp-playwright/Pages/CheckoutCompletePage.cs
using Microsoft.Playwright;

namespace PlaywrightFramework.Pages;

public class CheckoutCompletePage
{
    private readonly IPage _page;

    public CheckoutCompletePage(IPage page)
    {
        _page = page;
    }

    private ILocator Title => _page.Locator(".complete-header").First;

    public async Task<string> GetTitleTextAsync()
    {
        return await Title.TextContentAsync() ?? string.Empty;
    }
}

File: multiStructure/frameworks/ts-playwright/pages/checkout/checkoutCompletePage.ts
import { type Page, type Locator } from '@playwright/test';

export class CheckoutComplete {
  readonly page: Page;
  readonly title: Locator;

  constructor(page: Page) {
    this.page = page;
    this.title = page.locator('.complete-header');
  }

  async getTitleText(): Promise<string> {
    return (await this.title.first().textContent()) ?? '';
  }
}